pkgs/build-support/mkshell/default.nix

```diff
@@ -28,6 +28,49 @@ let
     "propagatedNativeBuildInputs"
     "shellHook"
   ];
+
+  docPathShellHook = let
+    unwrapCC = pkg: if pkg ? cc then unwrapCC pkg.cc else pkg;
+    docInputs = buildInputs ++ nativeBuildInputs
+                ++ propagatedBuildInputs ++ propagatedNativeBuildInputs
+                # grab the unwrapped cc, as wrappers may or may not
+                # retain doc outputs
+                ++ (lib.optional (stdenv ? cc) (unwrapCC stdenv.cc))
+                ++ stdenv.defaultBuildInputs ++ stdenv.defaultNativeBuildInputs
+                ++ stdenv.initialPath;
+    docDrvs = lib.pipe docInputs [
+      (lib.subtractLists stdenv.disallowedRequisites)
+      (builtins.filter lib.isDerivation)
+      lib.unique
+    ];
+    docPaths = type: lib.pipe docDrvs [
+      (map (lib.getOutput type))
+      (map (out: builtins.toPath "${out}/share/${type}"))
+      (builtins.filter builtins.pathExists)
+    ];
+    infoPaths = docPaths "info";
+    manPaths = docPaths "man";
+
+    # some packages neglect to build an info dir file
+    infoDirAddenda = stdenv.mkDerivation {
+      inherit (stdenv) system;
+      name = "${name}-info-dir-addenda";
+      passAsFile = ["buildCommand"];
+      buildCommand = ''
+        shopt -s nullglob
+        mkdir -p $out/share/info
+        for pkg in ${builtins.concatStringsSep " " infoPaths}; do
+          [[ -s $pks/share/info/dir ]] && continue || :
```

Comment:
Typo?

Reply:
you mean the colon?  it's like "true", I use it reflexively in such cases to avoid thinking "is this running under -e or not?"

Reply:
No, I am talking about $pks instead of $pkg

Reply:
oh, oops, thank you.